.github/workflows/actions.yml

```diff
@@ -27,23 +41,21 @@ jobs:
     - name: Lint with flake8
       run: |
         conda install flake8
-        git fetch origin ${{ github.event.pull_request.base.ref }}
-        echo $(git diff --name-only origin/${{ github.event.pull_request.base.ref }} | grep ".py$")
-        flake8 $(git diff --name-only origin/${{ github.event.pull_request.base.ref }} | grep ".py$")
+        flake8 ${{ needs.check-lint.outputs.output1 }}
     
     - name: Lint with black
       run: |
         conda install black
-        git fetch origin ${{ github.event.pull_request.base.ref }}
-        echo $(git diff --name-only origin/${{ github.event.pull_request.base.ref }} | grep ".py$")
-        black $(git diff --name-only origin/${{ github.event.pull_request.base.ref }} | grep ".py$") --check
+        black ${{ needs.check-lint.outputs.output1 }} --check
     
     - name: Type-check with myy
       run: |
         conda install mypy
         mypy src/
 
   tests:
+    needs: [check-lint]
+    if: needs.check-lint.outputs.output1 != ''
```

Comment:
Not sure we want to prevent tests from running, what if a configuration file changed that caused a failure?